src/main/java/ua/lviv/iot/part1/lab6/reposistory/MantisReposistory.java

```diff
@@ -0,0 +1,57 @@
+package ua.lviv.iot.part1.lab6.reposistory;
+
+import ua.lviv.iot.part1.lab6.model.Mantis;
+
+import java.util.HashMap;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.atomic.AtomicInteger;
+
+
+public class MantisReposistory {
+
+    private final Map<Integer, Mantis> mapOfMantis = new HashMap<>();
+    private final AtomicInteger availableId = new AtomicInteger();
+
+    public Integer getNextAvailableId() {
+
+        availableId.set(1);
+
+        while (mapOfMantis.containsKey(availableId.get())) {
+
+            availableId.incrementAndGet();
+        }
+        return availableId.get();
+    }
+
+    public Mantis getById(Integer mantisId) {
+        return mapOfMantis.get(mantisId);
+    }
+
+    public void postMantisToMap(Mantis mantis) {
```

Comment:
the repository has no idea about post. Post is a HTTP method and should not be used outside of Controller